lib/src/generate/build_impl.dart

```diff
@@ -375,15 +375,29 @@ class BuildImpl {
 
     // Skip the prompt if using this option.
     if (_deleteFilesByDefault) {
+      _logger.info('Deleting ${conflictingOutputs.length} declared outputs '
+          'which already existed on disk.');
       await deleteConflictingOutputs();
       return;
     }
 
     // Prompt the user to delete files that are declared as outputs.
-    stdout.writeln('\n\nFound ${conflictingOutputs.length} declared outputs '
+    _logger.warning('Found ${conflictingOutputs.length} declared outputs '
         'which already exist on disk. This is likely because the'
         '`$cacheDir` folder was deleted, or you are submitting generated '
         'files to your source repository.');
+
+    // If not in a standard terminal then we just exit, since there is no way
+    // for the user to provide a yes/no answer.
+    if (stdioType(stdin) != StdioType.TERMINAL) {
+      _logger.severe('Exiting since not in interactive mode and'
+          '`deleteFilesByDefault` was not set to true.');
+      exit(1);
```

Comment:
Could we throw a special exception here and turn it into an exit code at the `main` level?

See https://github.com/dart-lang/test/blob/master/lib/src/executable.dart#L152